app/components/inputs/PathInput.js

```diff
@@ -0,0 +1,56 @@
+const electron = require("electron");
+const dialog = electron.remote.dialog;
+const mainWindow = electron.remote.getCurrentWindow();
+const ipc = electron.ipcRenderer;
+
+import { PathButton } from "../buttons";
+
+class PathInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { path: "" };
+    this.selectDirectory = this.selectDirectory.bind(this);
+  }
+
+  componentDidMount() {
+    const self = this;
+    ipc.on("path", function (event, data) {
```

Comment:
This won't work here, since this listener will be bound forever. Try the following and see the problem:
- Start in advanced daemon mode
- Select a Daemon Data Dir
- Go to the other "tab" and select a cert file

You should see an exception due to setState below being called on an unmounted component.

One option to solve this, would be to move the whole file/directory selection logic into either an action or (probably better) a function that returns a promise. Then this promise gets resolved when the selection is done or rejected if canceled/errored and you use a `.once` listener so that only *that* particular call is triggered.

Reply:
I was originally trying to create a utility component that didn't need to be tied to redux state, but creating an action is the right thing - thanks. I'm from react-native land and the ipc main and renderer process are new to me. I'm guessing it's like a system thread and a ui thread

Reply:
You can still do it without going through redux by using a promise. Something like:
```javascript
const selectDir = () => new Promise((reject, resolve)) => {
  //...
  ipcRenderer.once('path'), (path) => resolve(path));
  dialog.showDirDialog(...);
  //...
}
```

Then use on the component by: 
```javascript
onSelectDir() {
  selectDir().then((path) => this.setState(...));
}
```

Reply:
@matheusd - I took a different approach with hopefully similar results. I am passing in an 'id' prop and using this id as a unique ipc event identifier. I am removing the listener upon the dismounting of the component to keep it tidy. I tested for multiple file/dir browsing components on the same screen and they seem to be functional. Let me know if you have more feedback upon review.